src/dtx/dtx_rpc.c

```diff
@@ -234,8 +234,16 @@ dtx_req_send(struct dtx_req_rec *drr, daos_epoch_t epoch)
 			din->di_flags.ca_arrays = NULL;
 		}
 
-		if (dra->dra_opc == DTX_REFRESH && DAOS_FAIL_CHECK(DAOS_DTX_RESYNC_DELAY)) {
-			rc = crt_req_set_timeout(req, 3);
+		if (dra->dra_opc == DTX_REFRESH) {
+			if (DAOS_FAIL_CHECK(DAOS_DTX_RESYNC_DELAY))
+				rc = crt_req_set_timeout(req, 3);
+			else
+				/*
+				 * If related DTX is committable, then it will be committed
+				 * within DTX_COMMIT_THRESHOLD_AGE time. So if need to wait
+				 * for longer, then just let related client to retry.
+				 */
+				rc = crt_req_set_timeout(req, DTX_COMMIT_THRESHOLD_AGE);
```

Comment:
min (rpc_timeout, DTX_COMMIT_THRESHOLD_AGE) ?

Reply:
In theory, you are right. I will consider to change it in subsequent patch.